### PR TITLE
Produce traces_service_graph_request_total service graph metric

### PR DESCRIPTION
**Pull request by aabmass**
This updates the servicegraph recipe to produce the standard `traces_service_graph_request_total` used by the servicegraph connector (and Tempo) documented in https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/servicegraphconnector#how-it-works.

- Updated the pipeline to remove some metric attributes we don't need
- Added a second round of `k8sattributes`, first to detect the server pod name from `server.address` then for the client pod name from `client.address`
- Updated the graphgen Go script to use the new metric name and added some filters

### CODE REVIEW

File: recipes/beyla-service-graph/collector-config.yaml [hunk 1]
```diff
@@ -53,47 +48,63 @@ spec:
...
-      k8sattributes:
+          - client.address
+      transform/serveraddress_to_k8spodip:
+        metric_statements:
+          - context: resource
+            statements:
+            - delete_key(attributes, "k8s.pod.name")
+            - set(attributes["k8s.pod.ip"], attributes["server.address"])
+      # Add k8s.pod.name
+      k8sattributes/1:
```
Comment (dashpole) [inline recipes/beyla-service-graph/collector-config.yaml:64]: this looks the same as k8sattributes/2.  Can you not use the same processor twice in a single pipeline?

File: recipes/beyla-service-graph/collector-config.yaml [hunk 2]
```diff
@@ -53,47 +48,63 @@ spec:
...
-          - key: k8s.pod.ip
-            from_attribute: server.address
-            action: insert
-      k8sattributes:
+          - client.address
+      transform/serveraddress_to_k8spodip:
+        metric_statements:
+          - context: resource
+            statements:
+            - delete_key(attributes, "k8s.pod.name")
```
Comment (dashpole) [inline recipes/beyla-service-graph/collector-config.yaml:61]: are we just re-doing work already done by beyla?  Why do we need to delete this attribute?

Comment (aabmass) [reply recipes/beyla-service-graph/collector-config.yaml:61]: I was hoping to remove the annotation being done by Beyla altogether once that bug is fixed. I could re-use this saved attribute in theory and remove the `k8sattributes/1`. Wdyt?

File: recipes/beyla-service-graph/collector-config.yaml (hunk 1 above)
Comment (aabmass) [reply recipes/beyla-service-graph/collector-config.yaml:64]: Unfortunately no it fails at runtime [see here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md#:~:text=The%20same%20name%20of%20the%20processor%20MUST%20NOT%20be%20referenced%20multiple%20times%20in%20the%20processors%20key%20of%20a%20single%20pipeline). I can add a comment explaining it

Comment (dashpole) [reply recipes/beyla-service-graph/collector-config.yaml:64]: A comment would be helpful

File: recipes/beyla-service-graph/collector-config.yaml (hunk 2 above)
Comment (dashpole) [reply recipes/beyla-service-graph/collector-config.yaml:61]: up to you.  A comment explaining what the plan is would be helpful for future reference

File: recipes/beyla-service-graph/collector-config.yaml (hunk 1 above)
Comment (aabmass) [reply recipes/beyla-service-graph/collector-config.yaml]: Updated PTAL

File: recipes/beyla-service-graph/graphgen/internal/query.go [hunk 3]
```diff
@@ -54,20 +66,30 @@ func QueryPrometheus(ctx context.Context, client api.Client, queryWindow time.Du
...
+func getQuery(queryArgs QueryArgs) string {
+	filters := addFilter(nil, "cluster", queryArgs.Cluster)
+	filters = addFilter(filters, "namespace", queryArgs.Namespace)
 	return fmt.Sprintf(
 		`sum by (
-			k8s_pod_ip, client_address, k8s_pod_name
+			server, client_address, client, server_address
 		) (
-			rate(http_servicegraph_calls_total[%s])
+			rate(traces_service_graph_request_total{%s}[%s]) != 0
```
Comment (aabmass) [inline recipes/beyla-service-graph/graphgen/internal/query.go:83]: The bug in #92 only happens for a short time after Beyla starts so I can filter out the zero valued rates

**Pull request merged**